build.gradle.kts

```diff
@@ -251,27 +252,26 @@ subprojects {
   }
 
   plugins.withType<BasePlugin> {
-    tasks.findByName("check")!!.apply {
-      if (hibernateProjects.contains(project.name)) {
-        testShardHibernate.dependsOn(this)
-      } else if (redisProjects.contains(project.name)) {
-        testShardRedis.dependsOn(this)
-      } else {
-        testShardNonHibernate.dependsOn(this)
-      }
+    if (hibernateProjects.contains(project.name)) {
+      testShardHibernate.configure { dependsOn("check") }
+    } else if (redisProjects.contains(project.name)) {
+      testShardRedis.configure { dependsOn("check") }
+    } else {
+      testShardNonHibernate.configure { dependsOn("check") }
+    }
 
+    tasks.named("check") {
       // Disable the default `detekt` task and enable `detektMain` which has type resolution enabled
       dependsOn(dependsOn.filterNot { name != "detekt" })
-      if (tasks.findByName("detektMain") != null) {
-        dependsOn("detektMain")
-      }
+      dependsOn(tasks.named { it == "detektMain" })
     }
   }
 
+  val configurationNames = setOf("kapt", "wire", "proto", "Proto")
   configurations.all {
     // Workaround the Gradle bug resolving multiplatform dependencies.
     // https://github.com/square/okio/issues/647
-    if (name.contains("kapt") || name.contains("wire") || name.contains("proto") || name.contains("Proto")) {
+    if (name in configurationNames) {
```

Comment:
This is not quite the same (`name.contains` vs `name ==`), but I don't know how relevant this copypasta is anymore anyway. Let's see if anything breaks!